articles/proofs-arguments-and-zero-knowledge/chapter_19.md

```diff
@@ -0,0 +1,174 @@
+---
+Title: Bird’s Eye View of Practical Arguments
+Status: In Progress
+Level: "6"
+---
+# Chapter 19: Bird’s Eye View of Practical Arguments
+
+![](attachments/taxonomy.png)
+
+## 19.1 A Taxonomy of SNARKs
+
+Outside of the **linear-PCP-based SNARKs**, most known **SNARKs** are obtained by combining some **IP**,  **MIP**, or **constant-round polynomial IOP** with a **polynomial commitment** scheme.
+
+For example:
+
+| Polynomial IOP approaches      | Polynomial Commitment Approaches                                                     | SNARK                                            |
+| ------------------------------ | ------------------------------------------------------------------------------------ | ------------------------------------------------ |
+| IPs                            | FRI-based (multilinear) polynomial commitments                                       | Virgo                                            |
+| IPs                            | Discrete-log-based (multilinear) polynomial commitments (Bulletproofs, Hyrax-commit) | Hyrax                                            |
+| IPs                            | KZG-based (multilinear) polynomial commitments                                       | zk-vSQL and Libra                                |
+| MIPs                           | Multilinear polynomial commitments                                                   | Spartan, Xiphos, Kopis, Brakedown, and Shockwave |
+| Constant-round polynomial IOPs | FRI-based (univariate) polynomial commitments                                        | Aurora, Fractal, and Redshift                    |
+| Constant-round polynomial IOPs | KZG-based (univariate) polynomial commitments                                        | Marlin                                           |
+| Constant-round polynomial IOP  | Ligero                                                                               | Ligero                                           |
+
+Beside that, the most popular variant of the SNARK derived from GGPR’s linear PCP is
+Groth16.
+
+**More SNARKs via composition**: On top of the taxonomy of SNARKs delineated above, one can take any two SNARKs designed via one of the above approaches, and compose them one or more times. Such compositions are growing increasingly popular and already yield state-of-the-art performance.
+
+## 19.2 Pros and Cons of the Approaches
+
+**Pros and cons of three first approach in section 10.6**.
+
+a fifth approach to argument design, based on commit-and-prove techniques (13.1) (not so related)
+
+(IP-based, MIP-based, and constant-round-polynomial-IOP-based)
+
+the information-theoretically secure protocol
+ +
+ any extractable polynomial commitment scheme
+ =
+
+  succinct argument
```

Comment:
why this becomes bigger like a heading?

Reply:
Hmm it is a MrSon's writing. So I think I need to modify it with him